docs/architecture/adr-047-handling-evidence-from-light-client.md

```diff
@@ -5,99 +5,99 @@
+
+Assuming one of the two providers is honest, that full node will then take the trace
+and extract out the relevant evidence and propogate it until it ends up on chain.
+
+*NOTE: We do not send evidence to other connected peers*
+
+*NOTE: The light client halts then and does not verify with any other witness*
+
+## Detailed Design
+
+The traces will have the following data structure: 
 
 ```go
-type ConflictingHeadersEvidence struct {
-  H1 types.SignedHeader
-  H2 types.SignedHeader
+type ConflictingHeadersTrace struct {
+  Headers []*types.SignedHeader
 }
 ```
 
-_Remark_: Theoretically, only the header, which differs from what a full node
-has, needs to be sent. But sending two headers a) makes evidence easily
-verifiable b) simplifies the light client, which does not have query each
-witness as to which header it possesses.
-
-When a full node receives the `ConflictingHeadersEvidence` evidence, it should
+When a full node receives a `ConflictingHeadersTrace`, it should
 a) validate it b) figure out if malicious behaviour is obvious (immediately
-slashable) or the fork accountability protocol needs to be started.
+slashable) or run the amnesia protocol.
 
 ### Validating headers
 
-Check both headers are valid (`ValidateBasic`), have the same height, and
-signed by 1/3+ of the validator set that the full node had at height
-`H1.Height`.
+Check headers are valid (`ValidateBasic`), are in order of ascending height, and do not exceed the `MaxTraceSize`.
 
-- Q: What if light client validator set is not equal to full node's validator
-  set (i.e. from full node's point of view both headers are not properly signed;
-  this includes the case where none of the two headers were committed on the
-  main chain)
+### Finding Block Bifurcation
 
-  Reject the evidence. It means light client is following a fork, but, hey, at
-  least it will halt.
+The node pulls the block ID's for the respective heights of the trace headers from its own block store.
 
-- Q: Don't we want to punish validators who signed something else even if they
-  have less or equal than 1/3?
+First it checks to see that the first header hash matches its first `BlockID` else it can discard it. 
 
-  No consensus so far. Ethan said no, Zarko said yes.
-  https://github.com/tendermint/spec/pull/71#discussion_r374210533
+If the last header hash matches the nodes last `BlockID` then it can also discard it on the assumption that a fork can not remerge and hence this is just a trace of valid headers. 
 
-### Figuring out if malicious behaviour is immediately slashable
+The node then continues to loop in descending order checking that the headers hash doesn't match it's own blockID for that height. Once it reaches the height that the block ID matches the hash it then sends the common header, the trusted header and the diverged header (common header is needed for lunatic evidence) to determine if the divergence is a real offense to the tendermint protocol or if it is just fabricated.
```

Comment:
sends where?

Reply:
I'm in the process of rewriting it again because we decided to change course. So would be great if you could review this #5273